outlier_screening.Rmd

```diff
@@ -1,37 +1,87 @@
+
+```{r boxplot} 
+boxplot(dt$cfratio)
+summary(dt$cfratio)
+upper_whisker = 3.327 + ((3.327-1.080) * 1.5)
+```
+
+## Review upper values 
+The upper whisker extends 1.5 times past the interquartile range, to `r upper_whisker`. 
+The values that lie beyond that are: 
+```{r upper_outliers} 
+high_cfratio = dt[cfratio>upper_whisker, .(country_code, Cases, Deaths, casepc, deathpc, cfratio)]
+kable(high_cfratio[order(cfratio)])
+hist(high_cfratio$cfratio)
+```
+
+Although these do represent high fatality ratios, many of them represent reasonable values. For example, the second-highest, Italy, had one of the worst early waves in the pandemic and this fatality ratio is very likely six months after they started recording cases/deaths. 
+
+The only value that I would treat like an outlier here is Yemen, which has a case-fatality ratio of 28.9. It looks highly likely that Yemen has undercounted cases in these reported numbers, because over 1 in 4 people have died after catching COVID-19 in their reported numbers (593 out of 2047). Because of this, we will drop this data point from our analysis. 
+
+```{r final_dataset} 
+# Generate final 6-month dataset that accounts for outlier screening 
+outlier_screened = copy(dt)
+outlier_screened = outlier_screened[!is.na(cfratio)]
+
+# Remove Yemen, because nearly over 1 in 4 people with COVID-19 have died (in reported numbers). 
+outlier_screened = outlier_screened[country_code!="YEM"]
+
+```
+
+## Review zeros/very low values 
+
+```{r boxplot2} 
+hist(outlier_screened$cfratio)
+boxplot(outlier_screened$cfratio)
+```
+There are also a large cluster of countries with values clustered at zero (N=`r outlier_screened[cfratio==0, .N]`), which would signify zero deaths. Review these to make sure these values are believable. 
+```{r low_values} 
+kable(outlier_screened[cfratio==0, .(country_code, Cases, Deaths, deathpc, casepc, pop_2019, clean_date, cfratio)])
+```
+
+There are a few NA values here where population was not available. We will remove these from our analysis. However, for the rest of the values, they all have very low reported case numbers and may have truly recorded zero deaths. Without more substantial evidence to signify a reporting failure, we will leave these zeros in for now. 
+
+```{r outlier_screening2} 
+outlier_screened = outlier_screened[!(is.na(pop_2019) | is.na(casepc) | is.na(deathpc))] # Drop anything that would cause your data to be calculated incorrectly. 
+write.csv(outlier_screened, "prepped_data/six_month_outlier_screened.csv", row.names=F)
+```
+
+# Finally, run initial analysis again and see if non-normality problem persists. 
+```{r final_check} 
+hist(outlier_screened$cfratio)
+boxplot(outlier_screened$cfratio)
+plot(lm(cfratio ~ overall, data=outlier_screened))
+```
+
+After this cleaning, we do see some evidence of non-constant variance, and our non-normality problem persists. We will need to turn to data transformations to address these issues. 
```

Comment:
How extreme are these? From the chat it sounds like these aren't enough to invalidate results. From lectures it sounds like there are often gray areas and since this is real world data we may be able to be flexible with the data and allow for a little more variance then if we were dealing with a rigorous science experiment

Reply:
To me, the variance doesn't seem like as much of a problem as the non-normality. There is a bit of spread in the variance from left to right, but it's not horrible. For normality though, our sample is definitely not normal. 

Then again, I don't know how much we can do about this. Normality is the least problematic of the four assumptions to violate, and the slides basically say "if you've got a big enough sample don't worry about it". So I think it's worth it to try a few transformations, but we might not be able to solve this one.